From b9d3a9f82d9bdec7e5772992e039432a490cc38c Mon Sep 17 00:00:00 2001
From: Electron752 <mzoran@crowfest.net>
Date: Sat, 15 Oct 2016 11:49:52 -0700
Subject: [PATCH 127/215] ARM64: Modify default config to get raspbian to boot
 (#1686)

1. Enable emulation of deprecated instructions.
2. Enable ARM 8.1 and 8.2 features which are not detected at runtime.
3. Switch the default governer to powersave.
4. Include the watchdog timer driver in the kernel image rather then a module.

Tested with raspbian-jessie 2016-09-23.
---
 arch/arm64/configs/bcmrpi3_defconfig | 18 +++++++++++-------
 1 file changed, 11 insertions(+), 7 deletions(-)

diff --git a/arch/arm64/configs/bcmrpi3_defconfig b/arch/arm64/configs/bcmrpi3_defconfig
index 1fcde8c..d7406f5 100644
--- a/arch/arm64/configs/bcmrpi3_defconfig
+++ b/arch/arm64/configs/bcmrpi3_defconfig
@@ -27,19 +27,23 @@ CONFIG_ARM64_VA_BITS=39
 CONFIG_SCHED_MC=y
 CONFIG_NR_CPUS=4
 CONFIG_HOTPLUG_CPU=y
+CONFIG_ARMV8_DEPRECATED=y
+CONFIG_SWP_EMULATION=y
+CONFIG_CP15_BARRIER_EMULATION=y
+CONFIG_SETEND_EMULATION=y
 
 #
 # ARMv8.1 architectural features
 #
-CONFIG_ARM64_HW_AFDBM=n
-CONFIG_ARM64_PAN=n
-CONFIG_ARM64_LSE_ATOMICS=n
-CONFIG_ARM64_VHE=n
+CONFIG_ARM64_HW_AFDBM=y
+CONFIG_ARM64_PAN=y
+CONFIG_ARM64_LSE_ATOMICS=y
+CONFIG_ARM64_VHE=y
 
 #
 # ARMv8.2 architectural features
 #
-CONFIG_ARM64_UAO=n
+CONFIG_ARM64_UAO=y
 CONFIG_ARM64_MODULE_CMODEL_LARGE=n
 CONFIG_RANDOMIZE_BASE=n
 
@@ -98,7 +102,7 @@ CONFIG_ZBOOT_ROM_TEXT=0x0
 CONFIG_ZBOOT_ROM_BSS=0x0
 CONFIG_CMDLINE="console=ttyAMA0,115200 kgdboc=ttyAMA0,115200 root=/dev/mmcblk0p2 rootfstype=ext4 rootwait"
 CONFIG_CPU_FREQ=y
-CONFIG_CPU_FREQ_DEFAULT_GOV_ONDEMAND=y
+CONFIG_CPU_FREQ_DEFAULT_GOV_POWERSAVE=y
 CONFIG_CPU_FREQ_GOV_PERFORMANCE=y
 CONFIG_CPU_FREQ_GOV_USERSPACE=y
 CONFIG_CPU_FREQ_GOV_ONDEMAND=y
@@ -687,7 +691,7 @@ CONFIG_SENSORS_SHTC1=m
 CONFIG_THERMAL=y
 CONFIG_THERMAL_BCM2835=y
 CONFIG_WATCHDOG=y
-CONFIG_BCM2835_WDT=m
+CONFIG_BCM2835_WDT=y
 CONFIG_UCB1400_CORE=m
 CONFIG_MFD_STMPE=y
 CONFIG_STMPE_SPI=y
-- 
1.9.1

